docs/DafnyRef/Refinement.md

```diff
@@ -92,32 +92,133 @@ TODO
 
 ## 21.4. Const declarations
 
-A parent `const` declaration may be refined by a refining `const` declaration
-if
+A child `const` declaration may refine a `const` declaration from a
+parent module if
 
 * the parent has no initialization,
 * the child has the same type as the parent, and
 * one or both of the following holds:
    * the child has an initializing expression
-   * the child is declared `ghost` and the parent is not `ghost`, or vice versa
+   * the child is declared `ghost` and the parent is not `ghost`, or vice versa.
 
-To indicate it is a refining declaration, a refining `const` declaration
-contains an ellipsis in this syntax:
-```grammar
-"const" { Attribute } CIdentType "..." [ ":=" Expression ]
-```
+A refining module can also introduce new `const` declarations that do
+not exist in the refinement parent.
 
 ## 21.5. Method declarations
 
-TODO
+A child `method` definition may refine a parent `method` declaration or
+definition to
+
+* provide a body missing in the parent,
+* replace the body of the parent with a new, semantically-compatible
+  body,
+* strengthen the postcondition of the parent method by adding one or more
+  `ensures` clauses,
+* provide a more deterministic version of a non-deterministic parent
+  body, or
+* amend the body of the parent method with additional statements.
```

Comment:
Do we say somewhere what the correctness criterion is for refinement? (If child can return `x` then parent can return `x`).  IIUC these are all variants of that general pattern.

Reply:
My understanding is that the correctness criteria for refinement are less strict than that, and are derived from that list. I may be wrong about that, though. @RustanLeino?